src/lib/libdylink.js

```diff
@@ -1313,11 +1313,11 @@ var LibraryDylink = {
 #if ASSERTIONS
     assert(lib, `Tried to dlsym() from an unopened handle: ${handle}`);
 #endif
-    if (!lib.exports.hasOwnProperty(symbol) || lib.exports[symbol].stub) {
+    newSymIndex = Object.keys(lib.exports).indexOf(symbol);
+    if (newSymIndex == -1 || lib.exports[symbol]?.stub) {
```

Comment:
```suggestion
    if (newSymIndex == -1 || lib.exports[symbol].stub) {
```
The code before does not seem to have checked that?